osgp/shared/shared/src/test/java/org/opensmartgridplatform/shared/application/config/scheduling/OsgpSchedulerTest.java

```diff
@@ -118,4 +133,27 @@ CronTrigger assertQuartzTrigger(
 
     return cronTrigger;
   }
+
+  SimpleTrigger assertQuartzSimpleTrigger(
+      final Class<? extends Job> jobClazz, final Integer delayInSeconds) throws SchedulerException {
+
+    verify(this.quartzScheduler).addJob(this.jobDetailArgumentCaptor.capture(), eq(true));
+    assertThat(this.jobDetailArgumentCaptor.getValue().getKey().getName())
+        .isEqualTo(jobClazz.getSimpleName());
+    verify(this.quartzScheduler)
+        .scheduleJob(
+            this.jobDetail2ArgumentCaptor.capture(),
+            this.jobTriggerSetArgumentCaptor.capture(),
+            eq(true));
+    assertThat(this.jobDetail2ArgumentCaptor.getValue())
+        .isEqualTo(this.jobDetailArgumentCaptor.getValue());
+    assertThat(this.jobTriggerSetArgumentCaptor.getValue().size()).isOne();
+    assertThat(this.jobTriggerSetArgumentCaptor.getValue().size()).isOne();
+    final SimpleTrigger simpleTrigger =
+        (SimpleTrigger) this.jobTriggerSetArgumentCaptor.getValue().iterator().next();
+    assertThat(simpleTrigger.getStartTime())
+        .isCloseTo(new DateTime().plusSeconds(delayInSeconds).toDate(), 100l);
+
+    return simpleTrigger;
```

Comment:
Seems to me this method could use some refactoring to increase readability? 

Furthermore assertions on `simpleTrigger` are performed in this method, but also in the calling `testCreateAndScheduleSimpleJobDataMap` method?

Reply:
I thought i could stand on the shoulder of giants, but i was wrong. ;-)
Refactored the testing methods